### ISSUE TITLE
Apptainer cannot download Docker image even when properly logged in

### ISSUE THREAD

**Issue by nicolamori** [apptainer/apptainer#2230]
### Version of Apptainer
```
$ apptainer --version
apptainer version 1.3.0
```
### Expected behavior
Apptainer should be able to pull images from a Gitlab Docker registry once logged in with credentials with appropriate permissions

### Actual behavior
In my setup using the same credentials that work for Docker, Apptainer cannot pull the image. Singularity 4.1.2 works properly.

### Steps to reproduce this behavior
```
$ docker login -u nicola_personal_rwregistry  git.herd.cloud.infn.it:5050
Password: 

$ docker pull git.herd.cloud.infn.it:5050/herd/herd-htc-frontend:latest
latest: Pulling from herd/herd-htc-frontend
Digest: sha256:5a42f7368706adc6eb105859a58dd7a8153358ba0c8d178857cd1d366fb680eb
Status: Image is up to date for git.herd.cloud.infn.it:5050/herd/herd-htc-frontend:latest
git.herd.cloud.infn.it:5050/herd/herd-htc-frontend:latest

$ apptainer registry login --username nicola_personal_rwregistry  docker://git.herd.cloud.infn.it:5050
Password / Token: 
INFO:    Token stored in /home/mori/.apptainer/remote.yaml

$ apptainer pull docker://git.herd.cloud.infn.it:5050/herd/herd-htc-frontend:latest
FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://git.herd.cloud.infn.it:5050/herd/herd-htc-frontend:latest: Requesting bearer token: invalid status code from registry 403 (Forbidden)
```

Removing Apptainer and installing Singularity I then get:

```
$ singularity --version
singularity-ce version 4.1.2

$ singularity registry login --username nicola_personal_rwregistry  docker://git.herd.cloud.infn.it:5050
Password / Token: 
INFO:    Token stored in /home/mori/.singularity/docker-config.json

$ singularity pull docker://git.herd.cloud.infn.it:5050/herd/herd-htc-frontend:latest
INFO:    Converting OCI blobs to SIF format
WARNING: 'nodev' mount option set on /tmp, it could be a source of failure during build process
INFO:    Starting build...
INFO:    Fetching OCI image...
21.6MiB / 265.2MiB [==================>--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 8 % 5.7 MiB/s 42s
242.5KiB / 17.3MiB [==>---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 1 % 64.4 KiB/s 4m30s
215.5KiB / 72.6MiB [>----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 0 % 57.3 KiB/s 21m34s
168.3KiB / 72.4MiB [>----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 0 % 44.7 KiB/s 27m33s
83.5KiB / 36.9MiB [>-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------] 0 % 22.2 KiB/s 28m19s
```

### What OS/distro are you running
```
$ cat /etc/os-release
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
```

### How did you install Apptainer
With `pacman` from the official `[extra]` Archlinux repository. I then removed Apptainer and installed Singularity from AUR.

**Comment by DrDaveD** [comment]
Is this a regression in 1.3?  It may be related to #2228.  Can you try to see if it works in apptainer-1.2.x?

**Comment by nicolamori** [comment]
It seems a regression since Apptainer 1.2.5 works:
```
$ apptainer --version
apptainer version 1.2.5

$ apptainer remote login --username nicola_personal_rwregistry  docker://git.herd.cloud.infn.it:5050
Password / Token: 
INFO:    Token stored in /home/mori/.apptainer/remote.yaml

$ apptainer pull docker://git.herd.cloud.infn.it:5050/herd/herd-htc-frontend:latest
INFO:    Converting OCI blobs to SIF format
WARNING: 'nodev' mount option set on /tmp, it could be a source of failure during build process
INFO:    Starting build...
Getting image source signatures
Copying blob 04d74f31e81c done  
Copying blob 8247731a7f82 done  
Copying blob 1eb9a6c0bdc7 done  
Copying blob 781d7e6ba786 [>-------------------------------------] 2.1MiB / 72.4MiB
Copying blob 2d473b07cdd5 skipped: already exists  
Copying blob 6ece7d43f5cb done  
Copying blob d75d786201fd done  
```
I also tried version 1.3.1 (didn't know there was a new release since Arch is currently shipping 1.3.0 and usually it's at bleeding edge so I did not check in the beginning) and it works as well:
```
$ apptainer --version
apptainer version 1.3.1

$ apptainer registry login --username nicola_personal_rwregistry  docker://git.herd.cloud.infn.it:5050
Password / Token: 
INFO:    Token stored in /home/mori/.apptainer/remote.yaml

$ apptainer pull docker://git.herd.cloud.infn.it:5050/herd/herd-htc-frontend:latest
INFO:    Converting OCI blobs to SIF format
WARNING: 'nodev' mount option set on /tmp, it could be a source of failure during build process
INFO:    Starting build...
Copying blob 1eb9a6c0bdc7 [--------------------------------------] 0.0b / 153.0b | 0.0 b/s
Copying blob 2d473b07cdd5 [--------------------------------------] 0.0b / 72.6MiB | 0.0 b/s
Copying blob 781d7e6ba786 [--------------------------------------] 0.0b / 72.4MiB | 0.0 b/s
Copying blob 6ece7d43f5cb [--------------------------------------] 0.0b / 402.0b | 0.0 b/s
Copying blob 04d74f31e81c [--------------------------------------] 0.0b / 2.0KiB | 0.0 b/s
Copying blob 8247731a7f82 [--------------------------------------] 0.0b / 858.0b | 0.0 b/s
```
So I'd say the regression in 1.3.0 has been fixed ion 1.3.1, this entry in the 1.3.1 changelog:
```
Fix problem where credentials locally stored with registry login command were not usable in some execution flows. Run registry login again with latest version to ensure credentials are stored correctly.
```
looks reasonable.

Thanks @DrDaveD for your suggestions!

**Comment by DrDaveD** [comment]
Oh good.  I'm sorry I forgot about that fix though.
